pkg/operator/standalone_starter.go

```diff
@@ -0,0 +1,119 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/api/sharedresource"
+	"github.com/openshift/cluster-storage-operator/pkg/csoclients"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/configobservation/configobservercontroller"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/csidriveroperator"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/defaultstorageclass"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/snapshotcrd"
+	"github.com/openshift/cluster-storage-operator/pkg/operator/vsphereproblemdetector"
+	"github.com/openshift/cluster-storage-operator/pkg/operatorclient"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/loglevel"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/managementstatecontroller"
+	"github.com/openshift/library-go/pkg/operator/status"
+	rbacv1 "k8s.io/api/rbac/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+)
+
+func startControllerStandAlone(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+	clients, err := csoclients.NewClients(controllerConfig, resync)
+	if err != nil {
+		return err
+	}
+
+	versionGetter := status.NewVersionGetter()
+	versionGetter.SetVersion("operator", status.VersionForOperatorFromEnv())
+
+	storageClassController := defaultstorageclass.NewController(
+		clients,
+		controllerConfig.EventRecorder,
+	)
+
+	countStorageClasses(storageClassController, clients)
+
+	// We dont' need this in hypershift guest or mgmt clusters
+	snapshotCRDController := snapshotcrd.NewController(
+		clients,
+		controllerConfig.EventRecorder,
+	)
```

Comment:
Can we remove the whole snapshotCRDController now?

IMO, CSISnapshotControllerOperator will get Degraded if it can't apply VolumeSnapshot CRDs. And it will overwrite any v1alpha1 CRDs.